client_cli/src/main.rs

```diff
@@ -892,6 +906,130 @@ mod asset {
             Ok(())
         }
     }
+
+    #[derive(StructOpt, Debug)]
+    pub struct SetStore {
+        /// AssetId for the Store asset (in form of `asset##account@domain_name')
+        #[structopt(long = "asset-id")]
+        pub asset_id: AssetId,
+        /// The key for the store value
+        #[structopt(long)]
+        pub key: Name,
+        /// The value to be associated with the key
+        #[structopt(long)]
+        pub value: String,
+        /// Interpret the `value` as a JSON string
+        #[structopt(long = "json", requires = "value")]
+        pub is_json: bool,
```

Comment:
Instead of specifying `--json` flag to control how `--value` should be parsed, I would suggest to use conflicting arguments `--value=<string>` and `--value-json=<json>`.